docs/architecture/adr-002-lite-client.md

```diff
@@ -0,0 +1,439 @@
+
+There are likely a few other instances of things the lite client is not validating that it in theory could but likely indicate some larger problem afoot that the client can't do anything about anyways. Hence we really only focus on the correctness of commits and validator sets and detecting forks!
+
+### API
+
+While the library should include pure functions for all the forms of
+verification, it may not be right to expose all this functionality, since it may
+then be possible to use it incorrectly (ie. a header verifies correctly but the state is actually expired).
+
+Thus, a stateful API should be provided for managing the trusted state of a lite
+node. The trusted state is:
+
+```rust
+pub struct TrustedState<H, V>
+where
+    H: Header,
+    V: ValidatorSet,
+{
+    pub last_header: H, // height H-1
+    pub validators: V,  // height H
+}
+```
+
+And the main method for verifying and updating the state should look like:
+
+```rust
+pub fn verify_sequential<H, V, C>(&mut self, 
+	now: Time, header: H, commit: C, validators: V, next_validators: V) -> Result<(), Error>
+where
+    H: Header,
+    V: ValidatorSet,
+    C: Commit,
+{
+```
+
+There is a corresponding `verify_skipping` method that is effectively the same but may also take a `trust_level`
+parameter to determine how much validator churn is acceptable (ie. can require some threshold greater than 1/3 
+of the trusted validators signing the new commit).
+
+Note this method takes a mutable reference to the TrustedState and may thus update it.
+It also contains all information necessary to validate and verify an update, including the current time,
+in order to check if the trusted state has expired.
+
+In the skipping case, if less than the trust level of trusted validators signed the commit, 
+an error should be returned that indicates to the caller that it should use bisection.
+For now, we leave the bisection functionality outside the core library, but in the future it may make 
+sense to bring it in with some generic way of "requesting" the bisecting headers (ie. it will need to work
+for both the lite node case, where the node can drive requests on demand, and the IBC case, where the blockchain
+must wait for the intermediate headers to be provided).
+
+## Status
+
+Proposed
+
+## Consequences
+
+### Positive
+
+- Implements the lite node!
+- Simple peering strategy
+- Clear separation between verification logic and the actual data types
+
+### Negative
+
+- Abstract traits requires more coding, more opportunity for bugs in trait implementation
+
+### Neutral
+
+- Certain validity checks are ommitted since they have little bearing
```

Comment:
Out of curuosity, what is being omitted?

Is it just checking that the proposer & round were valid? (If so, totally agreed with that being unnecessary)

Reply:
I outlined a few points in the `Validation` section above. Didn't mention round and proposer, but those are additional things that aren't checked. Mentioned things like LastCommitHash, BlockID.PartsHeader, and Time